Proton/Sources/ObjC/include/PRTextStorage.h

```diff
@@ -25,21 +25,35 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
-@protocol DefaultTextFormattingProviding;
-@protocol TextStorageDelegate;
-@class Attachment;
+NS_SWIFT_NAME(DefaultTextFormattingProviding)
+@protocol PRDefaultTextFormattingProviding<NSObject>
+@optional
+@property UIFont *font;
+@property NSMutableParagraphStyle *paragraphStyle;
+@property UIColor *textColor;
+@end
+
+@class PRTextStorage;
+
+@protocol TextStorageDelegate<NSObject>
+@required
+- (void) textStorage:(PRTextStorage *) textStorage didDelete: (NSTextAttachment *) attachment;
+@optional
```

Comment:
You declared these two methods optional but on caller side in `PRTextStorage` you are not checking with `respondsToSelector` before calling them

Reply:
I made these optional initially, but now thinking about that again, I can make it required. This is anyways internal implementation.